roles/cert_verify/tasks/check_endpoint.yml

```diff
@@ -0,0 +1,23 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Check for HTTPS in endpoint
+  ansible.builtin.assert:
+    that:
+      - "'https://' in cert_verify_endpoint_to_check.URL"
+    success_msg: "{{ cert_verify_endpoint_to_check.URL }} is HTTPS"
+    fail_msg: "{{ cert_verify_endpoint_to_check.URL }} is not HTTPS"
+  when: '"swift" not in cert_verify_endpoint_to_check.URL'
```

Comment:
same, `when: cert_verify_endpoint_to_check.URL is not match('swift')`

Also, care to put the condition right after the `name: Check for HTTPS in endpoint` ?